packages/medusa/src/api/admin/products/import/route.ts

```diff
@@ -2,27 +2,21 @@ import {
   AuthenticatedMedusaRequest,
   MedusaResponse,
 } from "@medusajs/framework/http"
-import { HttpTypes } from "@medusajs/framework/types"
-import { MedusaError } from "@medusajs/framework/utils"
+import type { HttpTypes } from "@medusajs/framework/types"
 import { importProductsWorkflow } from "@medusajs/core-flows"
+import type { AdminImportProductsType } from "../validators"
 
 export const POST = async (
-  req: AuthenticatedMedusaRequest<HttpTypes.AdminImportProductRequest>,
+  req: AuthenticatedMedusaRequest<AdminImportProductsType>,
   res: MedusaResponse<HttpTypes.AdminImportProductResponse>
 ) => {
-  const input = req.file as Express.Multer.File
-
-  if (!input) {
-    throw new MedusaError(
-      MedusaError.Types.INVALID_DATA,
-      "No file was uploaded for importing"
-    )
-  }
+  const fileProvider = req.scope.resolve("file")
+  const file = await fileProvider.getAsBuffer(req.body.filename)
 
   const { result, transaction } = await importProductsWorkflow(req.scope).run({
     input: {
-      filename: input.originalname,
-      fileContent: input.buffer.toString("utf-8"),
+      filename: req.body.originalname,
```

Comment:
`validatedbody`?

Reply:
Yup. Doing it here https://github.com/medusajs/medusa/pull/12328/files#diff-0d02e04e3c70c173d2c2ad5ca4e0133d7d9377243b624b68a058d3f1c7929a35R345-R351